torch/_inductor/scheduler.py

```diff
@@ -2457,21 +2437,38 @@ def can_fuse_vertical(
                 why("intermediate nodes between node1 & node2")
                 return False
 
-        # similar to can_inplace, if we are going to fuse a write subsequent to a read
-        # require that the indexing and size is the same
-        for write in node2.read_writes.writes:
-            if not isinstance(write, MemoryDep):
-                continue
-            for read in node1.read_writes.reads:
-                if write.name != self.mutation_renames.get(read.name, read.name):
-                    continue
+        return True
 
-                # bail on StarDep
-                if not self.fusable_read_and_write(read, write):
-                    why("fusing a write into a read with different indexing formula")
-                    return False
+    def fusable_weak_dep(
+        self, weak_dep: WeakDep, node1: BaseSchedulerNode, node2: BaseSchedulerNode
+    ) -> bool:
+        # A weak dep can be fused if and only if the fused operation acts inplace
+        # on the buffer being mutated. i.e. the same index is being read then mutated
```

Comment:
Doesn't it also need to be the only consumer of the buffer being mutated?

Reply:
I don't see why it would. The fused node `{op1, op2}`  can still have unmet weak deps on other nodes which must run before the fused node. For example you could have:
```
def fn(x):
    y = x + 2
    x.copy_(x - 2)
    return y, x
```
And we could fuse `x.copy_(x - 2)` into a single operation that has a WeakDep on y.

Reply:
Ah I think I see now, we can fuse the mutating node into one of many readers, the deps will be merged, and all the previous readers will still have to run before the combined node because the original deps on the mutating node will be respected. Sounds good.